lib/slack/rtm.ex

```diff
@@ -10,15 +10,22 @@ defmodule Slack.Rtm do
   @moduledoc false
 
   def start(token) do
-    url = Application.get_env(:slack, :url, "https://slack.com") <> "/api/rtm.start?token="
+    slack_url(token)
+    |> HTTPoison.get()
+    |> handle_response()
+  end
 
-    case HTTPoison.get(url <> token) do
-      {:ok, %HTTPoison.Response{body: body}} ->
-        case JSX.decode(body, [{:labels, :atom}]) do
-          {:ok, json}       -> {:ok, json}
-          {:error, reason}  -> {:error, %JSX.DecodeError{reason: reason, string: body}}
-        end
-      {:error, reason} -> {:error, reason}
+  defp handle_response({:ok, %HTTPoison.Response{body: body}}) do
+    case JSX.decode(body, [{:labels, :atom}]) do
```

Comment:
@BlakeWilliams I thought it would be better to explicitly handle the deserializing failure scenario, considering that slack will send you text with it rate limits you. This allows the user to handle this scenario explicitly rather than having to throw a try around `Slack.Bot.start/1`